controllers/storagecluster/reconcile.go

```diff
@@ -496,6 +497,38 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 			// mark operator upgradeable if and only if all storageclusters are ready
 			if r.clusters.AreOtherStorageClustersReady(instance) {
+				if instance.Spec.AllowRemoteStorageConsumers {
+					scList := &ocsv1alpha1.StorageConsumerList{}
+					err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: instance.Namespace})
+					if err != nil {
+						r.Log.Error(err, "Failed to list StorageConsumers")
+						return reconcile.Result{}, err
+					}
+					providerVersion, err := semver.Make(version.Version)
+					if err != nil {
+						r.Log.Error(err, "Failed to parse OCS Version")
+						return reconcile.Result{}, err
+					}
+					for idx := range scList.Items {
+						clientVersion, err := semver.Make(scList.Items[idx].Status.Client.OperatorVersion)
+						if err != nil {
+							r.Log.Error(err, "Failed to parse Client operator version")
+							return reconcile.Result{}, err
+						}
```

Comment:
I am not sure this is the desired way to handle issues with the client version. 
I would prefer to mark that missing information to block the upgrade but not to kill the reconciliation process.  
Of course we will need to log the issue

Reply:
done.